tests/testGaussianProcess.py

```diff
@@ -35,8 +35,495 @@
+        the correct data when it should.
+        """
+        rng = np.random.RandomState(112)
+        data = rng.random_sample((10,10))
+        kd = gp.KdTreeD()
+        kd.Initialize(data)
+
+        # test that an exception is thrown if you ask for a point with
+        # a negative index
+        with self.assertRaises(RuntimeError) as context:
+            kd.getData(-1)
+
+        # test that an exception is thrown if you ask for a point beyond
+        # the number of points stored in the tree
+        with self.assertRaises(RuntimeError) as context:
+            kd.getData(10)
+
+        # test that the correct values are returned when they should be
+        for ix in range(10):
+            vv = kd.getData(ix)
+            for iy in range(10):
+                self.assertAlmostEqual(data[ix][iy], vv[iy], places=10)
+
+    def testKdTreeNeighborExceptions(self):
+        """
+        This test will test that KdTree throws exceptions when you ask it for
+        nonsensical number of nearest neighbors.
+        """
+        rng = np.random.RandomState(112)
+        data = rng.random_sample((10,10))
+        kd = gp.KdTreeD()
+        kd.Initialize(data)
+        pt = rng.random_sample(10).astype(float)
+        neighdex = np.zeros((5), dtype=np.int32)
+        distances = np.zeros((5), dtype=float)
+
+        # ask for a negative number of neighbors
+        with self.assertRaises(RuntimeError) as context:
+            kd.findNeighbors(neighdex, distances, pt, -2)
+
+        # ask for zero neighbors
+        with self.assertRaises(RuntimeError) as context:
+            kd.findNeighbors(neighdex, distances, pt, 0)
+
+        # ask for more neighbors than you have data
+        with self.assertRaises(RuntimeError) as context:
+            kd.findNeighbors(neighdex, distances, pt, 11)
+
+        # try sending neighdex of wrong size
+        neighdex_bad = np.zeros((1), dtype=np.int32)
+        with self.assertRaises(RuntimeError) as context:
+            kd.findNeighbors(neighdex_bad, distances, pt, 5)
+
+        # try sending distances array of wrong size
+        distances_bad = np.zeros((1), dtype=float)
+        with self.assertRaises(RuntimeError) as context:
+            kd.findNeighbors(neighdex, distances_bad, pt, 5)
+
+        # run something that works
+        kd.findNeighbors(neighdex, distances, pt, 5)
+
+    def testKdTree(self):
+        """
+        This test will test the construction of KdTree in the pathological case
+        where many of the input data points are identical.
+        """
+        pp = 100
+        dd = 5
+        data = np.zeros((pp, dd), dtype=float)
+        tol = 1.0e-10
+
+        f = open(os.path.join(testPath, "data", "kd_test_data.sav"))
+        ff = f.readlines()
+        f.close()
+        for i in range(len(ff)):
+            s = ff[i].split()
+            for j in range(dd):
+                data[i][j] = float(s[j])
+
+        kd = gp.KdTreeD()
+        try:
```

Comment:
What's the purpose of these try/catch blocks? Are you expecting the `pex.Exception` to happen, or are you trying to intercept an error that should not happen and print it out? If the former, why not use the more standard `with self.assertRaises`, if the latter, shouldn't this be a test failure? If it's something non-standard then there should be a comment explaining it before the first try.